### ISSUE TITLE
Support association with a model has custom type primary key

### ISSUE THREAD

**Issue by sakuna63** [gfx/Android-Orma#368]
Example: 

If we have following model definitions

```java
@Table
class Parent {
    @PrimaryKey
    public CustomKey customKey;
}

@Table
class Child {
    @PrimaryKey
    public long id;

    @Column
    Parent parent;
}

class CustomKey {
    // ...
}

@StaticTypeAdapter(targetType = CustomKey.class, serializedType = String.class)
class CustomeKeyAdapter {
    // ...
}
```

Generated `AssociationDef` should use serialized `customKey` by CustomeKeyAdapter, but actually it attempts to use raw `customeKey` like a below

```java
class Child_Scheme {
    // ...
    public final AssociationDef<Child, Parent, Parent_Schema> parent;

    this.parent = new AssociationDef<Child, Parent, Parent_Schema> ... {
        // ...
      @NonNull
      @Override
      public String getSerialized(@NonNull Child model) {
          return model.parnet.customKey; // customKey should be serialized here by CustomeKeyAdpater!!!
      }
    // ...
    };
}
```

Whould you have any plan to support this feature?
Thanks.

**Comment by gfx** [comment]
This is definitely a bug around type adapters. I'm working in progress on it in #370.

**Comment by sakuna63** [comment]
Thanks!! 🙏

**Comment by gfx** [comment]
Released v4.1.1 including this fix. Thanks for the report! 🎆
